### PR TITLE
Swift package manager support

### PR DESCRIPTION
**Pull request by loudmouth**
This PR restructures the repo's directories to enable Swift Package Manager.

It also adds a build job to the travis matrix that ensures `swift build` is working. Unfortunately, SPM does not enable including other resources (i.e. bundled content) like Xcode so therefore `swift test` fails as cassettes cannot be found. Once support for bundling other resources with `swift build`, then we should update the travis script to run `swift test`.

Lastly, this fixes the travis builds that were broken on `master`.

Addresses #67

### CODE REVIEW

**Comment by loudmouth** [comment]
Only the iOS build has failed because of the infamous exit code 65 which is a known Xcode bug. Can somebody on the venmo org please restart the build? It should pass.

File: Readme.markdown
Comment (kiliankoe) [inline Readme.markdown]: This is probably the most minor of minor nitpicks ever, but the design of those two badges doesn't match up. How about using shields.io's custom badges for spm compatibility? 😊 E.g. https://img.shields.io/badge/SPM-compatible-brightgreen.svg

Comment (loudmouth) [reply Readme.markdown]: Nitpick all you want @kiliankoe  ;-)

Pushed with the badge you suggested. Thanks for the quick review!

**Comment by loudmouth** [comment]
Anything else that needs to be changed before merging?

**Review by eliperkins — APPROVED** [review]
This seems good to me! Thanks for looking into this.

Would you mind rebasing onto master to pickup the latest CI fixes?

**Comment by loudmouth** [comment]
Hey @eliperkins 

I've just rebased

**Comment by loudmouth** [comment]
Shoot...have an issue the iOS build in Travis...will have to fix when i'm back at a computer.

**Comment by loudmouth** [comment]
@eliperkins it looks like the error code on the iOS build is the infamous error code 65 and that job 
 of the build should just be restarted and it will probably pass.

**Comment by loudmouth** [comment]
@eliperkins just thought I'd check in on the status of this PR.

Hope you're doing well.

**Comment by dmiluski** [comment]
Hi @loudmouth, could you rebase once more? There appears to be an Interaction.swift conflict

**Comment by loudmouth** [comment]
Hey @dmiluski  and @eliperkins 

I'm really sorry it took me so long to get back to this—it's good to go now!

I had to update some code related to creating the test file for the upload test to pass on iOS, but everything is now green ✅ 

Any chance we can give this a merge and call it a day?

**Comment by loudmouth** [comment]
Just a note that we'll still need these two (duplicate) tickets in Swift Package Manager to be addressed for DVR to work with the swift build system on the command line:

[SR-2866](https://bugs.swift.org/browse/SR-2866)
[SR-4725](https://bugs.swift.org/browse/SR-4725)

Let's upvote there!

**Comment by loudmouth** [comment]
@eliperkins @kiliankoe @dmiluski 

I hope I don't become annoying in doing this, but i'd love to push forward this PR and then continue onto pushing for #80 after I rebase there.

This project is awesome and I use it to test the official [Contentful Swift SDK](https://github.com/contentful/contentful.swift), just wanna help keep it up-to-date and working for us all :-)

File: Scripts/travis-build-test.sh
Comment (eliperkins) [inline Scripts/travis-build-test.sh:6]: This is only running `build` and not `test` because of what you mentioned in this comment, correct? https://github.com/venmo/DVR/pull/73#issuecomment-396842789

File: Scripts/travis-build-test.sh
Comment (eliperkins) [inline Scripts/travis-build-test.sh:12]: Is this script taken from the Travis macOS config script? Is there a way we could shell out to that instead (so that we don't need to duplicate the code)?

File: Tests/DVRTests/SessionUploadTests.swift
Comment (eliperkins) [inline Tests/DVRTests/SessionUploadTests.swift:89]: I wonder if this should throw instead of crashing... Not a blocker for this PR!

File: .travis.yml
Comment (eliperkins) [inline .travis.yml:16]: Did this need to change? I think we want to prefer attribute-based selection of destinations, rather than names (which Travis could change from underneath us at any time).

**Review by eliperkins — APPROVED** [review]
LGTM, a few comments, but no blockers really.

File: Scripts/travis-build-test.sh
Comment (loudmouth) [reply Scripts/travis-build-test.sh:6]: yes exactly 👌

File: Tests/DVRTests/SessionUploadTests.swift
Comment (loudmouth) [reply Tests/DVRTests/SessionUploadTests.swift:89]: yeah, i just figured since it's in the testing suite it didn't matter too much. if you think it's ok for now, i'll leave it.

File: Scripts/travis-build-test.sh
Comment (loudmouth) [reply Scripts/travis-build-test.sh:12]: Hey, I'm not sure I understand. This is not a duplicate exactly of the script that was inlined int eh travis.yml previously. It's standard `xcodebuild` arguments, some of which consume environment variables which are set by Travis.

> Is there a way we could shell out to that instead?

The travis yml is calling this script, as opposed to inlining the `xcodebuild` command.

Am I understanding correctly?

File: .travis.yml
Comment (loudmouth) [reply .travis.yml:16]: I can try reverting this change and see how it goes 👍

File: Tests/DVRTests/SessionUploadTests.swift
Comment (eliperkins) [reply Tests/DVRTests/SessionUploadTests.swift:89]: Yeah, this is totally fine! Either way, the test suite would crash anyway.

File: Scripts/travis-build-test.sh
Comment (eliperkins) [reply Scripts/travis-build-test.sh:12]: Ah, my question was more around "is there a version of this command that comes in the Travis container that we can use instead of rewriting it ourselves?"

That is, we can supply these Travis-specific env vars in the .travis.yml, and Travis will generate an `xcodebuild` command for us. I _believe_ there's a `build.sh` script that is on the machine that's running the build, but I think it might be too late by the time our build has started to use this.

I don't think this should be a major blocker, but if we can figure out how to use the Travis env vars to generate an `xcodebuild` command rather than rewriting the command ourselves, we can save ourselves some headaches in the future.

File: .travis.yml
Comment (loudmouth) [reply .travis.yml:16]: so it turns out that both the platform and the architecture must be specified. the final command (with env variables filled in) is:

```bash
xcodebuild -jobs `sysctl -n hw.ncpu` test -project DVR.xcodeproj -scheme DVR-macOS -sdk macosx \
  -destination "platform=macOS,arch=x86_64" ONLY_ACTIVE_ARCH=YES CODE_SIGNING_IDENTITY="" CODE_SIGNING_REQUIRED=NO | xcpretty -c
```

I'm hoping Apple doesn't change the operating system name that often :-D

File: Scripts/travis-build-test.sh
Comment (eliperkins) [reply Scripts/travis-build-test.sh:12]: That said, I can't seem to find any documentation about this in the Travis docs. We can skip over this for now, unless you want to do any more digging or reverse engineering 😛

File: .travis.yml
Comment (eliperkins) [reply .travis.yml:16]: Awesome! I think `"platform=macOS,arch=x86_64"` is much better than `OS X`.

(especially when we need to do `"platform=macOS,arch=arm64"` down the line 😉 )

File: Scripts/travis-build-test.sh
Comment (loudmouth) [reply Scripts/travis-build-test.sh:12]: Ahh I see @eliperkins . I'm not too sure to be honest. I just googled around for a bundled build.sh for mac builds on travis but didn't find anything myself (it was a quick google to fair).

I'm not sure how different it is to delegate to a bash script, as you could see before my change the `xcodebuild` command was inlined right under `script:` in the yml file. The benefit of doing things this way is that we can have more fine grained control (i.e. if `SWIFT_BUILD` is set, then avoid running `xcodebuild`) and add more complexity to our build/test script. Of course we could have also done this inline in the yml file, but then it looks cluttered. It's just an approach I've taken on a lot of projects with travis, but I'm happy to revert and inline everything if you prefer. Just let me know.

The build is passing again, and I reverted the one env variable for the platform specification.

Comment (eliperkins) [reply Scripts/travis-build-test.sh:12]: All good with me! Let's keep this as it is in this PR.

My main concern here was duplicating the usage of the Travis-specific variables. If Travis is going to dictate the API for variables like `TRAVIS_XCODE_SCHEME` and such, it would be good to ensure that we use their build command rather than duplicate it ourself.

Given that we can't seem to find how they use these variables beyond what we see in the output, I think it's fine to just roll our own. 👍

Comment (loudmouth) [reply Scripts/travis-build-test.sh:12]: cool. i can investigate more as well and try to de-dup in my follow-up which i'm interested in #80 

always happy to improve CI setup

**Comment by loudmouth** [comment]
I think that's all the things. I'll leave it up to you guys at this point :-)

**Comment by loudmouth** [comment]
@eliperkins @kiliankoe @dmiluski any chance we can merge?

I wanna rebase #80 so that CI passes there.

**Pull request merged**
